src/test/run-make-fulldeps/coverage-reports-base/expected_show_coverage.abort.txt

```diff
@@ -0,0 +1,35 @@
+    1|       |#![feature(unwind_attributes)]
+    2|       |#![allow(unused_assignments)]
+    3|       |
+    4|       |#[unwind(aborts)]
+    5|       |fn might_abort(should_abort: bool) {
+    6|      4|    if should_abort {
+    7|      0|        println!("aborting...");
+    8|      0|        panic!("panics and aborts");
+    9|      4|    } else {
+   10|      4|        println!("Don't Panic");
+   11|      4|    }
+   12|      4|}
+   13|       |
+   14|       |fn main() -> Result<(),u8> {
+   15|      1|    let mut countdown = 10;
+   16|     11|    while countdown > 0 {
+   17|     10|        if countdown < 5 {
+   18|      4|            might_abort(false);
+   19|      6|        }
```

Comment:
This is actually correct.

The condition `countdown < 5` executed 10 times (10 loop iterations).

It evaluated to `true` 4 times, and executed the `might_abort()` call.

It skipped the body of the `might_abort()` call 6 times (hidden `else`).

I wanted to capture the coverage of the non-true condition. Say the condition was `countdown < 50`, which is always `true`. In that case, we wouldn't have a test for what happens if `might_abort()` is not called.

So counting the "hidden else" is important.

I can clarify this, but I should probably clarify it in a different test that's more focused on `if` blocks inside loops, and keep this test focused on testing the `Abort` terminator.

One side-note... The Span for the "hidden else" starts as a 0-length span right after the right brace, if I'm not mistaken. But code region counts versus line counts are confusing for 0-length regions, so I always expend them by 1 character, preferably to the right (in the typical case, after the brace) UNLESS it's at the end of the line, in which case I have to expand it to the left (which is the right brace). 

You can't see it because there are no more code regions on that line, and the code block ends at the semicolon after the `might_abort()` call.

If there was no semicolon (say, if might_abort() returned a value), then there would be multiple code regions on that line, and you would see the one character highlighted. I'm not sure that would help clarify things anymore than in it's existing state though.

Reply:
Yeah, I was able to infer what it was doing, I just found it confusing at first. Your explanation was helpful to understand _why_ doing this is a good idea, thanks.

Reply:
I went ahead and expanded this example, and added comments and notes to help explain the counter for the implicit `else`.

Reply:
Great news here! I changed how the Span is set for implicit `else` when lowering from AST, and it works! No more special case for `Goto`. Lots of related issues resolved.